### PR TITLE
Overlay translations (copy-locale and copy from ghost)

### PR DESCRIPTION
**Pull request by trickreich**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | ..
| Related issues/PRs | ..
| License | MIT
| Documentation PR | ..

#### What's in this PR?

Made translations for overlay configurable.

### CODE REVIEW

**Comment by trickreich** [comment]
@danrot @wachterjohannes What do you guys think about this?

**Comment by wachterjohannes** [comment]
@trickreich please add changelog and readme

File: src/Sulu/Bundle/ContentBundle/Resources/public/js/components/copy-locale-overlay/main.js [hunk 1]
```diff
@@ -80,6 +90,27 @@ define(function() {
...
+         * Otherwise use the default translation.
+         *
+         * @param {String} id
+         *
+         * @return {String}
+         */
+        getTranslation = function(id) {
+            var translationKey = translations[id];
+            if (this.translations
+                && this.translations.copyLocaleOverlay
```
Comment (danrot) [inline src/Sulu/Bundle/ContentBundle/Resources/public/js/components/copy-locale-overlay/main.js]: How would that value be passed?

**Comment by danrot** [comment]
Isn't that a little bit complicated? Why not using the options of the component instead?

**Comment by trickreich** [comment]
Because this overlay is started inside, some other needed data also passed via context.

**Comment by trickreich** [comment]
@wachterjohannes Can you have a look once more? Which readme do you mean?

File: src/Sulu/Bundle/ContentBundle/Resources/public/js/components/copy-locale-overlay/main.js (hunk 1 above)
Comment (danrot) [inline src/Sulu/Bundle/ContentBundle/Resources/public/js/components/copy-locale-overlay/main.js]: @trickreich Still waiting for an explanation here 😉

Comment (trickreich) [inline src/Sulu/Bundle/ContentBundle/Resources/public/js/components/copy-locale-overlay/main.js]: Sorry, told it in the comment:
Other needed data also passed via context.

**Comment by trickreich** [comment]
@danrot Code is refactored

File: src/Sulu/Bundle/ContentBundle/Resources/public/js/components/copy-locale-overlay/main.js [hunk 2]
```diff
@@ -227,13 +248,20 @@ define(function() {
...
+         * Start overlay with selects to choose src and dest languages.
+         *
+         * @param {Object} newTranslations
+         *
+         * @return {*}
          */
-        startCopyLocalesOverlay: function() {
+        startCopyLocalesOverlay: function(newTranslations) {
+            // overwrite translations
+            translations = this.sandbox.util.extend(true, {}, translations, newTranslations);
```
Comment (danrot) [inline src/Sulu/Bundle/ContentBundle/Resources/public/js/components/copy-locale-overlay/main.js]: Doesn't that change the existing defaults? So when it is called later without the `newTranslation` parameter, the defaults will be overridden?

File: src/Sulu/Bundle/ContentBundle/Resources/public/js/components/open-ghost-overlay/main.js [hunk 3]
```diff
@@ -102,17 +122,25 @@ define(function() {
...
+         * @param {Object} newTranslations
+         *
          * @returns {Object}
          */
-        openGhost: function(item) {
+        openGhost: function(item, newTranslations) {
             var def = this.sandbox.data.deferred();
 
+            // overwrite translations
+            translations = this.sandbox.util.extend(true, {}, translations, newTranslations);
```
Comment (danrot) [inline src/Sulu/Bundle/ContentBundle/Resources/public/js/components/open-ghost-overlay/main.js]: Same as above

**Comment by trickreich** [comment]
@danrot I've changed the marked lines.

**Comment by trickreich** [comment]
@danrot FR translations are added

**Pull request merged**